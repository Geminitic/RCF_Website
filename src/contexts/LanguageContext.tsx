import React, { createContext, useContext, useState, ReactNode } from 'react';

interface Language {
  code: 'en' | 'ar';
  name: string;
  direction: 'ltr' | 'rtl';
}

interface LanguageContextType {
  currentLanguage: Language;
  setLanguage: (_language: Language) => void;
  t: (_key: string, _enText: string, _arText: string) => string;
}

const languages: Language[] = [
  { code: 'en', name: 'English', direction: 'ltr' },
  { code: 'ar', name: 'العربية', direction: 'rtl' },
];

const LanguageContext = createContext<LanguageContextType | undefined>(
  undefined
);

export const useLanguage = () => {
  const context = useContext(LanguageContext);
  if (!context) {
    throw new Error('useLanguage must be used within a LanguageProvider');
  }
  return context;
};

interface LanguageProviderProps {
  children: ReactNode;
}

export const LanguageProvider: React.FC<LanguageProviderProps> = ({
  children,
}) => {
  const [currentLanguage, setCurrentLanguage] = useState<Language>(
    languages[0]
  );

  const setLanguage = (language: Language) => {
    setCurrentLanguage(language);
    document.documentElement.dir = language.direction;
    document.documentElement.lang = language.code;
  };

  const t = (_: string, enText: string, arText: string): string => {
    return currentLanguage.code === 'ar' ? arText : enText;
  };

  return (
    <LanguageContext.Provider value={{ currentLanguage, setLanguage, t }}>
      {children}
    </LanguageContext.Provider>
  );
};

export { languages };
