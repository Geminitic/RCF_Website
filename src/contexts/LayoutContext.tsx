/* eslint-disable react-refresh/only-export-components */
import React from 'react'

export const LayoutContext = React.createContext({})

export function LayoutProvider({ children }: { children: React.ReactNode }) {
  return <LayoutContext.Provider value={{}}>{children}</LayoutContext.Provider>
}
