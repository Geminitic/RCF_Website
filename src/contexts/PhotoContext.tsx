import React, { createContext, useContext, useState, ReactNode, useMemo } from 'react';
import { photoList } from '../data/photoList';

interface Photo {
  id: string;
  url: string;
  title: string;
  titleAr: string;
  description: string;
  descriptionAr: string;
  location: string;
  locationAr: string;
  category: string;
  uploadedBy: string;
  uploadDate: string;
  approved: boolean;
  featured: boolean;
  likes: number;
  comments: { id: string; name: string; text: string }[];
}

interface PhotoContextType {
  photos: Photo[];
  uploadedCount: number;
  targetCount: number;
  addPhoto: (
    photo: Omit<Photo, 'id' | 'uploadDate' | 'approved'>,
    approved?: boolean
  ) => void;
  likePhoto: (id: string) => void;
  addComment: (id: string, name: string, text: string) => void;
  approvePhoto: (id: string) => void;
  deletePhoto: (id: string) => void;
  pendingPhotos: Photo[];
  featuredPhotos: Photo[];
  communityPhotos: Photo[];
  allPhotos: Photo[];
}

const PhotoContext = createContext<PhotoContextType | undefined>(undefined);

export const usePhoto = () => {
  const context = useContext(PhotoContext);
  if (!context) {
    throw new Error('usePhoto must be used within a PhotoProvider');
  }
  return context;
};

interface PhotoProviderProps {
  children: ReactNode;
}

export const PhotoProvider: React.FC<PhotoProviderProps> = ({ children }) => {
  const [photos, setPhotos] = useState<Photo[]>(() => {
    const base: Photo[] = [
    // Real community photos
    {
      id: '1',
      url: '/484134928_122145258254471410_1133353559349890810_n.jpg',
      title: 'Community Gathering',
      titleAr: 'تجمع مجتمعي',
      description: 'A vibrant community event bringing people together',
      descriptionAr: 'فعالية مجتمعية نابضة بالحياة تجمع الناس معاً',
      location: 'Aleppo',
      locationAr: 'حلب',
      category: 'community',
      uploadedBy: 'RCF Team',
      uploadDate: '2024-01-15',
      approved: true,
      featured: true
    },
    {
      id: '2',
      url: '/WhatsApp Image 2025-06-17 at 12.41.33 AM.jpeg',
      title: 'Cultural Festival',
      titleAr: 'مهرجان ثقافي',
      description: 'Celebrating local heritage and traditions',
      descriptionAr: 'الاحتفال بالتراث والتقاليد السورية',
      location: 'Damascus',
      locationAr: 'دمشق',
      category: 'culture',
      uploadedBy: 'Community Member',
      uploadDate: '2024-01-10',
      approved: true,
      featured: true
    },
    {
      id: '3',
      url: '/WhatsApp Image 2025-06-19 at 12.35.09 PM.jpeg',
      title: 'Youth Engagement',
      titleAr: 'مشاركة الشباب',
      description: 'Young people participating in community activities',
      descriptionAr: 'الشباب يشاركون في الأنشطة المجتمعية',
      location: 'Homs',
      locationAr: 'حمص',
      category: 'youth',
      uploadedBy: 'Youth Coordinator',
      uploadDate: '2024-01-08',
      approved: true,
      featured: true
    },
    {
      id: '4',
      url: '/WhatsApp Image 2025-06-19 at 12.35.10 PM.jpeg',
      title: 'Educational Workshop',
      titleAr: 'ورشة تعليمية',
      description: 'Learning and skill development session',
      descriptionAr: 'جلسة تعلم وتطوير المهارات',
      location: 'Latakia',
      locationAr: 'اللاذقية',
      category: 'education',
      uploadedBy: 'Education Team',
      uploadDate: '2024-01-05',
      approved: true,
      featured: true
    },
    // New community photos from attachments
    {
      id: '5',
      url: '/504435580_688802243919719_1259884751197443936_n.jpeg',
      title: 'Community Outreach Program',
      titleAr: 'برنامج التوعية المجتمعية',
      description: 'Volunteers engaging with local community members',
      descriptionAr: 'متطوعون يتفاعلون مع أعضاء المجتمع المحلي',
      location: 'Edmonton, Canada',
      locationAr: 'إدمونتون، كندا',
      category: 'community',
      uploadedBy: 'Rhizome Canada',
      uploadDate: '2024-01-20',
      approved: true,
      featured: true
    },
    {
      id: '6',
      url: '/WhatsApp Image 2025-06-17 at 12.35.13 AM (2).jpeg',
      title: 'Community Support Initiative',
      titleAr: 'مبادرة الدعم المجتمعي',
      description: 'Team members providing assistance and support',
      descriptionAr: 'أعضاء الفريق يقدمون المساعدة والدعم',
      location: 'Local Region',
      locationAr: 'المنطقة',
      category: 'community',
      uploadedBy: 'RCF Team',
      uploadDate: '2024-01-18',
      approved: true,
      featured: true
    },
    {
      id: '7',
      url: '/WhatsApp Image 2025-06-17 at 12.35.14 AM.jpeg',
      title: 'Environmental Project',
      titleAr: 'مشروع بيئي',
      description: 'Community members working on environmental restoration',
      descriptionAr: 'أعضاء المجتمع يعملون على الترميم البيئي',
      location: 'Local Region',
      locationAr: 'المنطقة',
      category: 'community',
      uploadedBy: 'Environmental Team',
      uploadDate: '2024-01-16',
      approved: true,
      featured: true
    },
    {
      id: '8',
      url: '/WhatsApp Image 2025-06-17 at 12.35.14 AM (1).jpeg',
      title: 'Tree Planting Initiative',
      titleAr: 'مبادرة زراعة الأشجار',
      description: 'Volunteers planting trees for community green spaces',
      descriptionAr: 'متطوعون يزرعون الأشجار للمساحات الخضراء المجتمعية',
      location: 'Local Region',
      locationAr: 'المنطقة',
      category: 'community',
      uploadedBy: 'Green Team',
      uploadDate: '2024-01-14',
      approved: true,
      featured: true
    },
    {
      id: '9',
      url: '/WhatsApp Image 2025-06-17 at 12.35.14 AM (2).jpeg',
      title: 'Community Garden Project',
      titleAr: 'مشروع الحديقة المجتمعية',
      description: 'Building sustainable community gardens together',
      descriptionAr: 'بناء حدائق مجتمعية مستدامة معاً',
      location: 'Local Region',
      locationAr: 'المنطقة',
      category: 'community',
      uploadedBy: 'Garden Team',
      uploadDate: '2024-01-12',
      approved: true,
      featured: true
    },
    {
      id: '10',
      url: '/Untitled design (1).jpg',
      title: 'Community Workshop',
      titleAr: 'ورشة مجتمعية',
      description: 'Hands-on training session for local residents',
      descriptionAr: 'جلسة تدريب عملية للسكان المحليين',
      location: 'Toronto, Canada',
      locationAr: 'تورونتو، كندا',
      category: 'community',
      uploadedBy: 'RCF Team',
      uploadDate: '2024-01-22',
      approved: true,
      featured: true
    },
    {
      id: '11',
      url: '/WhatsApp Image 2025-06-17 at 12.39.55 AM (1).jpeg',
      title: 'Arts Event',
      titleAr: 'فعالية فنية',
      description: 'Showcasing creative work from youth members',
      descriptionAr: 'عرض أعمال إبداعية من أعضاء الشباب',
      location: 'Damascus',
      locationAr: 'دمشق',
      category: 'culture',
      uploadedBy: 'Arts Collective',
      uploadDate: '2024-01-21',
      approved: true,
      featured: true
    },
    {
      id: '12',
      url: '/WhatsApp Image 2025-06-19 at 12.35.09 PM copy.jpeg',
      title: 'Food Distribution',
      titleAr: 'توزيع الطعام',
      description: 'Volunteers distributing meals to families',
      descriptionAr: 'متطوعون يوزعون وجبات للأسر',
      location: 'Aleppo',
      locationAr: 'حلب',
      category: 'community',
      uploadedBy: 'Relief Team',
      uploadDate: '2024-01-19',
      approved: true,
      featured: true
    },
    {
      id: '13',
      url: '/WhatsApp Image 2025-06-19 at 12.35.09 PM copy copy.jpeg',
      title: 'Community Meeting',
      titleAr: 'اجتماع مجتمعي',
      description: 'Planning upcoming neighbourhood projects',
      descriptionAr: 'التخطيط لمشاريع الحي القادمة',
      location: 'Edmonton, Canada',
      locationAr: 'إدمونتون، كندا',
      category: 'community',
      uploadedBy: 'Rhizome Canada',
      uploadDate: '2024-01-17',
      approved: true,
      featured: true
    }
    ];

    const existingUrls = new Set(base.map(p => p.url));
    const extras: Photo[] = photoList
      .filter(url => !existingUrls.has(url))
      .map((url, index) => ({
        id: `auto-${index}`,
        url,
        title: 'Community Photo',
        titleAr: 'صورة المجتمع',
        description: 'Shared by the community.',
        descriptionAr: 'صورة تمت مشاركتها من المجتمع.',
        location: 'Unknown',
        locationAr: 'غير معروف',
        category: 'community',
        uploadedBy: 'Community',
        uploadDate: new Date().toISOString(),
        approved: true,
        featured: false
      }));

    const initial = [...base, ...extras].map((p) => ({
      ...p,
      likes: parseInt(
        typeof window !== 'undefined'
          ? localStorage.getItem(`likes-${p.id}`) || '0'
          : '0'
      ),
      comments: typeof window !== 'undefined'
        ? JSON.parse(localStorage.getItem(`comments-${p.id}`) || '[]')
        : []
    }));

    return initial;
  });

  // Total number of stories the community aims to collect
  const targetCount = 5000;

  // Count only the photos that have been approved
  const uploadedCount = photos.filter(p => p.approved).length;

  const addPhoto = (
    photoData: Omit<Photo, 'id' | 'uploadDate' | 'approved'>,
    approved = false
  ) => {
    const newPhoto: Photo = {
      ...photoData,
      id: Date.now().toString(),
      uploadDate: new Date().toISOString(),
      approved,
      likes: 0,
      comments: []
    };
    setPhotos(prev => [newPhoto, ...prev]);
  };

  const likePhoto = (id: string) => {
    if (typeof window === 'undefined') return;
    if (localStorage.getItem(`liked-${id}`)) return;
    setPhotos(prev =>
      prev.map(p =>
        p.id === id ? { ...p, likes: p.likes + 1 } : p
      )
    );
    const current = parseInt(localStorage.getItem(`likes-${id}`) || '0') + 1;
    localStorage.setItem(`likes-${id}`, String(current));
    localStorage.setItem(`liked-${id}`, '1');
  };

  const addComment = (id: string, name: string, text: string) => {
    if (typeof window === 'undefined') return;
    const comment = { id: Date.now().toString(), name, text };
    setPhotos(prev =>
      prev.map(p =>
        p.id === id ? { ...p, comments: [...p.comments, comment] } : p
      )
    );
    const stored = JSON.parse(localStorage.getItem(`comments-${id}`) || '[]');
    stored.push(comment);
    localStorage.setItem(`comments-${id}`, JSON.stringify(stored));
  };

  const approvePhoto = (id: string) => {
    setPhotos(prev => prev.map(photo =>
      photo.id === id ? { ...photo, approved: true } : photo
    ));
  };

  const deletePhoto = (id: string) => {
    setPhotos(prev => prev.filter(photo => photo.id !== id));
  };

  const featuredPhotos = photos.filter(photo => photo.featured && photo.approved);
  const communityPhotos = photos.filter(photo => photo.approved);
  const pendingPhotos = photos.filter(photo => !photo.approved);
  
  // Generate additional random photos for the full wall
  const randomPhotos = useMemo(() => {
    const generated = [];
    for (let i = 10; i <= 50; i++) {
      generated.push({
        id: `random-${i}`,
        url: `https://source.unsplash.com/600x600/?syria&sig=${i}`,
        title: `Community Story ${i}`,
        titleAr: `قصة المجتمع ${i}`,
        description: 'A story from our growing community network',
        descriptionAr: 'قصة من شبكة مجتمعنا المتنامية',
        location: 'Various Locations',
        locationAr: 'مواقع متنوعة',
        category: 'community',
        uploadedBy: 'Community Member',
        uploadDate: new Date(Date.now() - Math.random() * 30 * 24 * 60 * 60 * 1000).toISOString(),
        approved: true,
        featured: false,
        likes: 0,
        comments: []
      });
    }
    return generated;
  }, []);

  const allPhotos = useMemo(() => [...communityPhotos, ...randomPhotos], [communityPhotos, randomPhotos]);

  return (
    <PhotoContext.Provider value={{
      photos,
      uploadedCount,
      targetCount,
      addPhoto,
      likePhoto,
      addComment,
      approvePhoto,
      deletePhoto,
      pendingPhotos,
      featuredPhotos,
      communityPhotos,
      allPhotos
    }}>
      {children}
    </PhotoContext.Provider>
  );
};
