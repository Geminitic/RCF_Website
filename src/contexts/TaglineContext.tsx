import React, { createContext, useContext, useState, ReactNode } from 'react';

export interface BilingualText {
  en: string;
  ar: string;
}

interface TaglineContextType {
  title: BilingualText;
  subtitle: BilingualText;
  setTagline: (title: BilingualText, subtitle: BilingualText) => void;
}

const defaultTitle: BilingualText = {
  en: 'Community Canvas',
  ar: 'لوحة المجتمع'
};

const defaultSubtitle: BilingualText = {
  en: 'A vibrant tapestry of stories, moments, and memories that celebrate our global Syrian community.',
  ar: 'نسيج نابض من القصص واللحظات والذكريات التي تحتفي بمجتمعنا السوري العالمي.'
};

const TaglineContext = createContext<TaglineContextType | undefined>(undefined);

export const useTagline = () => {
  const ctx = useContext(TaglineContext);
  if (!ctx) {
    throw new Error('useTagline must be used within a TaglineProvider');
  }
  return ctx;
};

interface Props {
  children: ReactNode;
}

export const TaglineProvider: React.FC<Props> = ({ children }) => {
  const [title, setTitle] = useState<BilingualText>(() => {
    const stored = localStorage.getItem('tagline-title');
    return stored ? JSON.parse(stored) : defaultTitle;
  });
  const [subtitle, setSubtitle] = useState<BilingualText>(() => {
    const stored = localStorage.getItem('tagline-subtitle');
    return stored ? JSON.parse(stored) : defaultSubtitle;
  });

  const setTagline = (t: BilingualText, s: BilingualText) => {
    setTitle(t);
    setSubtitle(s);
    localStorage.setItem('tagline-title', JSON.stringify(t));
    localStorage.setItem('tagline-subtitle', JSON.stringify(s));
  };

  return (
    <TaglineContext.Provider value={{ title, subtitle, setTagline }}>
      {children}
    </TaglineContext.Provider>
  );
};
