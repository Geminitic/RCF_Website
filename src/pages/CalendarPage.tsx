import React, { useState, useEffect, useCallback } from 'react';
import BigCalendar from '../components/calendar/BigCalendar';
import { motion } from 'framer-motion';
import { 
  Calendar as CalendarIcon, 
  Clock, 
  MapPin, 
  Users, 
  Filter,
  Grid,
  List,
  Plus,
  ExternalLink,
  AlertCircle,
  Star,
  Tag,
  Loader
} from 'lucide-react';
import { useLanguage } from '../contexts/LanguageContext';
import type { CalendarEvent } from '../types';

const initialEvents: CalendarEvent[] = [];

const CalendarPage: React.FC = () => {
  const { t, currentLanguage } = useLanguage();
  const [viewMode, setViewMode] = useState<'calendar' | 'list'>('list');
  const [selectedCategory, setSelectedCategory] = useState('all');
  const [events, setEvents] = useState<CalendarEvent[]>([]);
  const [loading, setLoading] = useState(true);
  const [error, setError] = useState<string | null>(null);
  const [lastUpdated, setLastUpdated] = useState<string | null>(null);

  const categories = [
    { key: 'all', en: 'All Events', ar: 'جميع الفعاليات', color: 'bg-gray-500' },
    { key: 'funding', en: 'Funding', ar: 'التمويل', color: 'bg-green-500' },
    { key: 'workshop', en: 'Workshops', ar: 'ورش العمل', color: 'bg-blue-500' },
    { key: 'conference', en: 'Conferences', ar: 'المؤتمرات', color: 'bg-purple-500' },
    { key: 'cultural', en: 'Cultural', ar: 'ثقافية', color: 'bg-pink-500' },
    { key: 'educational', en: 'Educational', ar: 'تعليمية', color: 'bg-orange-500' },
    { key: 'networking', en: 'Networking', ar: 'تواصل', color: 'bg-indigo-500' },
    { key: 'general', en: 'General', ar: 'عام', color: 'bg-gray-500' }
  ];


  const fetchEvents = useCallback(async () => {
    setLoading(true);
    setError(null);
    
    try {
      const response = await fetch('/api/events');

      if (!response.ok) {
        throw new Error(`HTTP error! status: ${response.status}`);
      }

      const data = await response.json();
      
      if (data.events) {
        // Merge scraped events with initial events
        const allEvents = [...initialEvents, ...data.events];
        setEvents(allEvents);
        setLastUpdated(data.metadata?.lastUpdated || new Date().toISOString());
      } else {
        // Fallback to just scraped events if no metadata
        const scrapedEvents = Array.isArray(data) ? data : [];
        setEvents([...initialEvents, ...scrapedEvents]);
        setLastUpdated(new Date().toISOString());
      }
    } catch (err) {
      console.error('Failed to fetch calendar events:', err);
      try {
        const fallbackRes = await fetch('/sample-events.json');
        if (fallbackRes.ok) {
          const fallbackData: CalendarEvent[] = await fallbackRes.json();
          setEvents(fallbackData);
        } else {
          setEvents(initialEvents);
        }
      } catch {
        setEvents(initialEvents);
      }
      setError('Failed to load live events. Showing sample events.');
    } finally {
      setLoading(false);
    }
  }, []);

  useEffect(() => {
    fetchEvents();
  }, [fetchEvents]);

  const filteredEvents = selectedCategory === 'all' 
    ? events 
    : events.filter(event => event.category === selectedCategory);

  // react-big-calendar handles navigation and month calculations internally

  const renderCalendarView = () => {
    const bigCalendarEvents = filteredEvents.map(e => ({
      title: t('event-title', e.title, e.titleAr),
      start: e.date ? new Date(e.date) : new Date(),
      end: e.date ? new Date(e.date) : new Date(),
      allDay: true,
    }));

    return (
      <BigCalendar events={bigCalendarEvents} language={currentLanguage.code} />
    );
  };

  const renderListView = () => {
    if (loading) {
      return (
        <div className="flex items-center justify-center py-12">
          <Loader className="h-8 w-8 animate-spin text-indigo-600 mr-3" />
          <span className="text-lg text-stone-600">Loading events...</span>
        </div>
      );
    }

    if (filteredEvents.length === 0) {
      return (
        <div className="text-center py-12">
          <CalendarIcon className="h-16 w-16 text-stone-400 mx-auto mb-4" />
          <h3 className="text-xl font-semibold text-stone-600 mb-2">No events found</h3>
          <p className="text-stone-500">Try adjusting your filters or check back later.</p>
        </div>
      );
    }

    return (
      <div className="space-y-4">
        {filteredEvents.map((event, index) => {
          const category = categories.find(cat => cat.key === event.category);
          return (
            <motion.div
              key={event.id}
              initial={{ opacity: 0, y: 20 }}
              animate={{ opacity: 1, y: 0 }}
              transition={{ duration: 0.5, delay: index * 0.05 }}
              className="bg-white rounded-xl p-6 shadow-lg hover:shadow-xl transition-shadow border-l-4"
              style={{ borderLeftColor: category?.color?.replace('bg-', '#') || '#6b7280' }}
            >
              <div className="flex items-start justify-between mb-4">
                <div className="flex-1">
                  <div className="flex items-center mb-2">
                    <div className={`w-3 h-3 rounded-full mr-3 ${category?.color || 'bg-gray-500'}`} />
                    <span className={`text-xs font-medium text-stone-500 uppercase tracking-wide ${currentLanguage.code === 'ar' ? 'font-arabic' : ''}`}>
                      {t(`category-${event.category}`, event.category, event.category)}
                    </span>
                    {event.priority && event.priority > 3 && (
                      <Star className="h-4 w-4 text-yellow-500 ml-2" />
                    )}
                    {event.deadline && (
                      <AlertCircle className="h-4 w-4 text-red-500 ml-2" />
                    )}
                  </div>
                  <h3 className={`text-xl font-bold text-stone-900 mb-2 ${currentLanguage.code === 'ar' ? 'font-arabic' : ''}`}>
                    {t('event-title', event.title, event.titleAr)}
                  </h3>
                  {event.description && (
                    <p className={`text-stone-600 mb-4 ${currentLanguage.code === 'ar' ? 'font-arabic' : ''}`}>
                      {t('event-description', event.description, event.descriptionAr)}
                    </p>
                  )}
                </div>
                {event.registrationRequired && (
                  <button className="ml-4 px-4 py-2 bg-indigo-600 text-white rounded-lg hover:bg-indigo-700 transition-colors flex items-center">
                    <span className={`mr-2 ${currentLanguage.code === 'ar' ? 'font-arabic' : ''}`}>
                      {t('register', 'Register', 'سجل')}
                    </span>
                    <ExternalLink className="h-4 w-4" />
                  </button>
                )}
              </div>

              <div className="grid md:grid-cols-2 lg:grid-cols-4 gap-4 text-sm text-stone-600 mb-4">
                {event.date && (
                  <div className="flex items-center">
                    <CalendarIcon className="h-4 w-4 mr-2" />
                    <span>{new Date(event.date).toLocaleDateString()}</span>
                  </div>
                )}
                {event.time && (
                  <div className="flex items-center">
                    <Clock className="h-4 w-4 mr-2" />
                    <span className={currentLanguage.code === 'ar' ? 'font-arabic' : ''}>
                      {t('event-time', event.time, event.timeAr)}
                      {event.duration && ` (${t('event-duration', event.duration, event.durationAr)})`}
                    </span>
                  </div>
                )}
                {event.location && (
                  <div className="flex items-center">
                    <MapPin className="h-4 w-4 mr-2" />
                    <span className={currentLanguage.code === 'ar' ? 'font-arabic' : ''}>
                      {t('event-location', event.location, event.locationAr)}
                    </span>
                  </div>
                )}
                {event.attendees && event.maxAttendees && (
                  <div className="flex items-center">
                    <Users className="h-4 w-4 mr-2" />
                    <span>
                      {event.attendees}/{event.maxAttendees} {t('attendees', 'attendees', 'حاضر')}
                    </span>
                  </div>
                )}
              </div>

              {event.tags && event.tags.length > 0 && (
                <div className="flex flex-wrap gap-2 mb-4">
                  {event.tags.slice(0, 5).map((tag, tagIndex) => (
                    <span
                      key={tagIndex}
                      className="px-2 py-1 bg-stone-100 text-stone-600 text-xs rounded-full flex items-center"
                    >
                      <Tag className="h-3 w-3 mr-1" />
                      {tag}
                    </span>
                  ))}
                </div>
              )}

              <div className="flex items-center justify-between pt-4 border-t border-stone-200">
                <div className="flex items-center space-x-4">
                  <span className={`text-sm text-stone-500 ${currentLanguage.code === 'ar' ? 'font-arabic' : ''}`}>
                    {t('organized-by', 'Organized by', 'نظمت من قبل')} {t('event-organizer', event.organizer, event.organizerAr)}
                  </span>
                  {event.isOnline && (
                    <span className="px-2 py-1 bg-blue-100 text-blue-800 text-xs rounded-full">
                      {t('online-event', 'Online Event', 'فعالية عبر الإنترنت')}
                    </span>
                  )}
                  {event.deadline && (
                    <span className="px-2 py-1 bg-red-100 text-red-800 text-xs rounded-full">
                      {t('deadline', 'Deadline', 'موعد نهائي')}
                    </span>
                  )}
                </div>
                {event.link && (
                  <a
                    href={event.link}
                    target="_blank"
                    rel="noopener noreferrer"
                    className="text-indigo-600 hover:text-indigo-700 text-sm font-medium flex items-center"
                  >
                    {t('view-details', 'View Details', 'عرض التفاصيل')}
                    <ExternalLink className="h-4 w-4 ml-1" />
                  </a>
                )}
              </div>
            </motion.div>
          );
        })}
      </div>
    );
  };

  return (
    <div className="min-h-screen bg-gradient-to-br from-stone-50 to-indigo-50 pt-16">
      {/* Hero Section */}
      <section className="py-20 bg-gradient-to-r from-indigo-600 to-indigo-700 text-white">
        <div className="max-w-7xl mx-auto px-4 sm:px-6 lg:px-8">
          <motion.div
            initial={{ opacity: 0, y: 30 }}
            animate={{ opacity: 1, y: 0 }}
            transition={{ duration: 0.8 }}
            className={`text-center ${currentLanguage.code === 'ar' ? 'font-arabic' : ''}`}
          >
            <h1 className="text-5xl font-bold mb-6">
              {t('calendar-title', 'Events & Opportunities', 'الفعاليات والفرص')}
            </h1>
            <p className="text-xl text-indigo-100 mb-8 max-w-3xl mx-auto">
              {t(
                'calendar-subtitle',
                'Discover live events, workshops, and opportunities from Syrian organizations and partners worldwide.',
                'اكتشف الفعاليات المباشرة وورش العمل والفرص من المنظمات السورية والشركاء حول العالم.'
              )}
            </p>

            {error && (
              <div className="max-w-md mx-auto mb-6 p-4 bg-yellow-100 border border-yellow-300 rounded-lg text-yellow-800">
                <div className="flex items-center">
                  <AlertCircle className="h-5 w-5 mr-2" />
                  <span className="text-sm">{error}</span>
                </div>
              </div>
            )}

            {lastUpdated && (
              <div className="text-sm text-indigo-200 mb-6">
                {t('last-updated', 'Last updated', 'آخر تحديث')}: {new Date(lastUpdated).toLocaleString()}
              </div>
            )}

            <div className="flex flex-col sm:flex-row gap-4 justify-center">
              <button 
                onClick={fetchEvents}
                disabled={loading}
                className="px-6 py-3 bg-white text-indigo-600 font-semibold rounded-lg hover:bg-indigo-50 transition-colors flex items-center justify-center disabled:opacity-50"
              >
                {loading ? (
                  <Loader className="h-5 w-5 animate-spin mr-2" />
                ) : (
                  <Plus className="h-5 w-5 mr-2" />
                )}
                {t('refresh-events', 'Refresh Events', 'تحديث الفعاليات')}
              </button>
              <button className="px-6 py-3 bg-indigo-800 text-white font-semibold rounded-lg hover:bg-indigo-900 transition-colors">
                {t('subscribe-calendar', 'Subscribe to Calendar', 'اشترك في التقويم')}
              </button>
            </div>
          </motion.div>
        </div>
      </section>

      {/* Controls */}
      <section className="py-8 bg-white border-b border-stone-200">
        <div className="max-w-7xl mx-auto px-4 sm:px-6 lg:px-8">
          <div className="flex flex-col lg:flex-row gap-4 items-center justify-between">
            {/* View Toggle */}
            <div className="flex items-center bg-stone-100 rounded-lg p-1">
              <button
                onClick={() => setViewMode('calendar')}
                className={`flex items-center px-4 py-2 rounded-md font-medium transition-colors ${
                  viewMode === 'calendar'
                    ? 'bg-white text-indigo-600 shadow-sm'
                    : 'text-stone-600 hover:text-stone-900'
                }`}
              >
                <Grid className="h-4 w-4 mr-2" />
                {t('calendar-view', 'Calendar', 'التقويم')}
              </button>
              <button
                onClick={() => setViewMode('list')}
                className={`flex items-center px-4 py-2 rounded-md font-medium transition-colors ${
                  viewMode === 'list'
                    ? 'bg-white text-indigo-600 shadow-sm'
                    : 'text-stone-600 hover:text-stone-900'
                }`}
              >
                <List className="h-4 w-4 mr-2" />
                {t('list-view', 'List', 'القائمة')}
              </button>
            </div>

            {/* Month Navigation handled by BigCalendar */}

            {/* Category Filter */}
            <div className="flex items-center space-x-2">
              <Filter className="h-5 w-5 text-stone-600" />
              <select
                value={selectedCategory}
                onChange={(e) => setSelectedCategory(e.target.value)}
                className={`px-4 py-2 border border-stone-300 rounded-lg focus:ring-2 focus:ring-indigo-500 focus:border-indigo-500 ${
                  currentLanguage.code === 'ar' ? 'font-arabic' : ''
                }`}
              >
                {categories.map((category) => (
                  <option key={category.key} value={category.key}>
                    {t(`category-${category.key}`, category.en, category.ar)}
                  </option>
                ))}
              </select>
            </div>
          </div>

          {/* Event Count */}
          <div className="mt-4 text-center text-stone-600">
            {t('showing-events', 'Showing', 'عرض')} {filteredEvents.length} {t('events', 'events', 'فعالية')}
            {selectedCategory !== 'all' && (
              <span> {t('in-category', 'in category', 'في فئة')} "{t(`category-${selectedCategory}`, selectedCategory, selectedCategory)}"</span>
            )}
          </div>
        </div>
      </section>

      {/* Calendar/List Content */}
      <section className="py-12">
        <div className="max-w-7xl mx-auto px-4 sm:px-6 lg:px-8">
          {viewMode === 'calendar' ? renderCalendarView() : renderListView()}
        </div>
      </section>
    </div>
  );
};

export default CalendarPage;
