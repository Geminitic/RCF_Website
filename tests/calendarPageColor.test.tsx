import { render, waitFor, screen } from '@testing-library/react';
import React from 'react';
import { vi } from 'vitest';
import CalendarPage from '../src/pages/CalendarPage';
import { LanguageProvider } from '../src/contexts/LanguageContext';

const mockEvents = [
  { id: '1', title: 'Test Event', titleAr: 'اختبار', category: 'funding' }
];

vi.stubGlobal(
  'fetch',
  vi.fn(() =>
    Promise.resolve({
      ok: true,
      json: () =>
        Promise.resolve({ events: mockEvents, metadata: { lastUpdated: '2025-01-01T00:00:00.000Z' } }),
    })
  ) as any
);

vi.useFakeTimers();

vi.mock('framer-motion', async () => {
  const actual = await vi.importActual<any>('framer-motion');
  return {
    ...actual,
    motion: {
      div: (props: any) => <div {...props} />,
    },
  };
});

vi.useRealTimers();

describe('CalendarPage color styles', () => {
  it('renders event with hex border color', async () => {
    const { container } = render(
      <LanguageProvider>
        <CalendarPage />
      </LanguageProvider>
    );
    await waitFor(() => expect(screen.getByText('Test Event')).toBeInTheDocument());

    const eventDiv = container.querySelector('.border-l-4');
    expect(eventDiv).toHaveStyle({ borderLeftColor: '#22c55e' });
    expect(eventDiv).toMatchSnapshot();
  });
});
