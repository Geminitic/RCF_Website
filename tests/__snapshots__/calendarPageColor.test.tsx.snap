// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CalendarPage color styles > renders event with hex border color 1`] = `
<div
  animate="[object Object]"
  class="bg-white rounded-xl p-6 shadow-lg hover:shadow-xl transition-shadow border-l-4"
  initial="[object Object]"
  style="border-left-color: rgb(34, 197, 94);"
  transition="[object Object]"
>
  <div
    class="flex items-start justify-between mb-4"
  >
    <div
      class="flex-1"
    >
      <div
        class="flex items-center mb-2"
      >
        <div
          class="w-3 h-3 rounded-full mr-3"
          style="background-color: rgb(34, 197, 94);"
        />
        <span
          class="text-xs font-medium text-stone-500 uppercase tracking-wide "
        >
          funding
        </span>
      </div>
      <h3
        class="text-xl font-bold text-stone-900 mb-2 "
      >
        Test Event
      </h3>
    </div>
  </div>
  <div
    class="grid md:grid-cols-2 lg:grid-cols-4 gap-4 text-sm text-stone-600 mb-4"
  />
  <div
    class="flex items-center justify-between pt-4 border-t border-stone-200"
  >
    <div
      class="flex items-center space-x-4"
    >
      <span
        class="text-sm text-stone-500 "
      >
        Organized by
         
      </span>
    </div>
  </div>
</div>
`;
